### PR TITLE
Reorganization of Terraform to Handle Modules for Multiple Account Management

### LINKED ISSUE #4: Best Practice: Account Topologies
What are the best practices when setting up an AWS organization in terms of [account topologies](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_getting-started_concepts.html)? For example I know the root and payer account of an organization must be created manually and should be kept secure and isolated from the operational accounts.  

Consideration might include the [aggregation of logs ](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-sharing-logs.html)and how best to consolidate alerting and monitoring at the org level. 

A terraform organizations module should, if possible, should be able create 1 or many accounts with horizontal organizational units and service control policies.

### PR DESCRIPTION
**Pull request by knechtionscoding**
feat: setting up each part as a module
feat: configure for github action automation

Solves #5 

Reorganized with a few things in mind.

1. Using modules here for each part I think will help make things make sense in the terraform without too much deviation from terraform norms
2. Will allow others to use these directly if they want.
3. Will allow us to more easily test and manage them.
4. Allow us to connect between the terraform a little better.

### CODE REVIEW

**Comment by knechtionscoding** [comment]
@jameshounshell @kanr Any thoughts? I think this will allow us to use count and pass in all kinds of things.

It also helps work towards solving #4 as we can setup organizational units and then accounts inside of that.

File: modules/account-creation/main.tf [hunk 1]
```diff
@@ -1,15 +1,9 @@
-provider "aws" {
-  region = "us-east-1"
-}
-
 # this must be a unique email address
 # this will be used to perform the root user password reset when cleaning up (deleting) the account
 # (or use something like a gmail alias, ex: <username>+aws.orgs.dev@gmail.com)
```
Comment (kanr) [inline modules/account-creation/main.tf:3]: I like the use of the gmail dot blindness!

**Comment by kanr** [comment]
Yeah this makes sense, Creation and Config is a good modularization. Not a whole lot to this part. Another thing to add to Config could be[ Budgets.](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/budgets_budget)

**Comment by knechtionscoding** [comment]
I agree. Budgets, and maybe locking down regions? Or do we want to do regional lock downs in connection with the service map?

**Comment by kanr** [comment]
I think after I work through #3 I will be able to say. I did plan on locking down to only specific regions it makes. 

It would be nice to design a module(s) that will scale up as the need for more accounts develops in a new organization. I would probably start with 2, (not including the root account) so that centralized logging can be established along with the development environment.

File: modules/account-creation/main.tf (hunk 1 above)
Comment (knechtionscoding) [reply modules/account-creation/main.tf:3]: All @jameshounshell idea.

**Comment by knechtionscoding** [comment]
> I think after I work through #3 I will be able to say. I did plan on locking down to only specific regions it makes.
> 
> It would be nice to design a module(s) that will scale up as the need for more accounts develops in a new organization. I would probably start with 2, (not including the root account) so that centralized logging can be established along with the development environment.

So I'm thinking that the module for account creation will take OUs and Account emails.

Basically create a data structure that ties those together. I.e. define your whole account layout via json.

Then we just iterate through and create the OUs, Accounts, and apply labels. Then tie the configuration to labels.

**Pull request closed without merging**